pkg/controller/istio/istio_controller_test.go

```diff
@@ -303,4 +309,477 @@ var _ = Describe("Istio controller tests", func() {
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			Expect(cli.Create(ctx, fc)).NotTo(HaveOccurred())
+
+			customDSCP := numorstring.DSCPFromInt(10)
+			istioCR.Spec.DSCPMark = &customDSCP
+			Expect(cli.Update(ctx, istioCR)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Verify FelixConfiguration has custom DSCP mark
+			updatedFC := &crdv1.FelixConfiguration{}
+			Expect(cli.Get(ctx, types.NamespacedName{Name: "default"}, updatedFC)).NotTo(HaveOccurred())
+			Expect(updatedFC.Spec.IstioDSCPMark).NotTo(BeNil())
+			Expect(updatedFC.Spec.IstioDSCPMark.ToUint8()).To(Equal(uint8(10)))
+			Expect(updatedFC.Annotations[istio.IstioOperatorAnnotationDSCP]).To(Equal("10"))
+		})
+
+		It("should detect user modification of IstioAmbientMode in FelixConfiguration", func() {
+			// Create FelixConfiguration with mismatched annotation and spec
+			userModifiedMode := "UserModified"
+			fc := &crdv1.FelixConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+					Annotations: map[string]string{
+						istio.IstioOperatorAnnotationMode: "Enabled",
+					},
+				},
+				Spec: crdv1.FelixConfigurationSpec{
+					IstioAmbientMode: &userModifiedMode,
+				},
+			}
+			Expect(cli.Create(ctx, fc)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("felixconfig IstioAmbientMode modified by user"))
+		})
+
+		It("should detect user modification of IstioDSCPMark in FelixConfiguration", func() {
+			// Create FelixConfiguration with mismatched annotation and spec
+			userModifiedDSCP := numorstring.DSCPFromInt(50)
+			fc := &crdv1.FelixConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+					Annotations: map[string]string{
+						istio.IstioOperatorAnnotationDSCP: "23",
+					},
+				},
+				Spec: crdv1.FelixConfigurationSpec{
+					IstioDSCPMark: &userModifiedDSCP,
+				},
+			}
+			Expect(cli.Create(ctx, fc)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("felixconfig IstioDSCPMark modified by user"))
+		})
+
+		It("should clear FelixConfiguration on deletion", func() {
+			// Create FelixConfiguration with Istio settings
```

Comment:
Would be a better test if instead of pre-creating the FelixConfig, we just had the controller set these values and then clear them (to verify the actual logic to set the fields is reversible)